File: cicte-app/src/app/services/plan.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable, throwError } from 'rxjs';
import { map, retry } from 'rxjs/operators';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root'
})

export class PlanService {
  API_URL = environment.API_URL;
  constructor(public http: HttpClient) {
  }
  getQuery( query: string, request: any) {
    const url = `${this.API_URL}${ query }`;
    const headers = new HttpHeaders({
      'Content-Type': 'application/json'
    });
    const body = JSON.stringify(request);
    console.log('Paramentros enviados: ' + body );
    return this.http.post(url, body, { headers: headers });
  }

  getPlan(plan: any) {
    return this.getQuery('plan/getPlan', plan)
                  .pipe( map(data => data['plan']));
  }
  getPlanById(plan: any) {
    return this.getQuery('plan/getPlanById', plan)
                  .pipe( map(data => data['plan']));
  }
  getPlanByName(plan: any) {
    return this.getQuery('plan/getPlanByName', plan)
                  .pipe( map(data => data['plan']));
  }
  addPlan(plan: any) {
    return this.getQuery( 'plan/addPlan', plan )
                  .pipe( map(data => data));
  }
  updatePlan(plan: any) {
    return this.getQuery( 'plan/updatePlan', plan )
                  .pipe( map(data => data));
  }
  deletePlan(plan: any) {
    return this.getQuery('plan/deletePlan', plan)
                  .pipe( map(data => data));
  }
}

File: rest-api/src/com/espe/cicte/ws/service/PlanService.java
package com.espe.cicte.ws.service;

import javax.ws.rs.Consumes;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

import com.espe.cicte.ws.db.IlusionDB;
import com.espe.cicte.ws.db.PlanDB;
import com.espe.cicte.ws.dto.PlanRequest;
import com.espe.cicte.ws.dto.PlanResponse;

@Path("/cicte/plan")
public class PlanService {

	private static PlanDB PlanDB = new PlanDB();
	
	@POST
	@Path("/addPlan")
	@Consumes({ MediaType.APPLICATION_JSON })
	@Produces({ MediaType.APPLICATION_JSON })
	public PlanResponse addPlan(PlanRequest request) {
		PlanResponse response = new PlanResponse();

		response = PlanDB.addPlan(request);
		
		return response;
	}
	
	@POST
	@Path("/getPlan")
	@Consumes({ MediaType.APPLICATION_JSON })
	@Produces({ MediaType.APPLICATION_JSON })
	public PlanResponse getPlan(PlanRequest request) {
		PlanResponse response = new PlanResponse();

		response = PlanDB.getPlan();
	
		return response;
	}
	
	@POST
	@Path("/getPlanByName")
	@Consumes({ MediaType.APPLICATION_JSON })
	@Produces({ MediaType.APPLICATION_JSON })
	public PlanResponse getPlanName(PlanRequest request) {
		PlanResponse response = new PlanResponse();

		response = PlanDB.getPlanByName(request);
	
		return response;
	}
	
	@POST
	@Path("/getPlanById")
	@Consumes({ MediaType.APPLICATION_JSON })
	@Produces({ MediaType.APPLICATION_JSON })
	public PlanResponse getPlanById(PlanRequest request) {
		PlanResponse response = new PlanResponse();

		response = PlanDB.getPlanById(request);
	
		return response;
	}
	
	@POST
	@Path("/updatePlan")
	@Consumes({ MediaType.APPLICATION_JSON })
	@Produces({ MediaType.APPLICATION_JSON })
	public PlanResponse updatePlan(PlanRequest request) {
		PlanResponse response = new PlanResponse();

		response = PlanDB.updatePlan(request);

		return response;
	}
	
	@POST
	@Path("/deletePlan")
	@Consumes({ MediaType.APPLICATION_JSON })
	@Produces({ MediaType.APPLICATION_JSON })
	public PlanResponse deletePlan(PlanRequest request) {
		PlanResponse response = new PlanResponse();

		response = PlanDB.deletePlan(request);

		return response;
	}
}
